src/main/java/org/opensearch/knn/index/mapper/ModelFieldMapper.java

```diff
@@ -50,7 +51,7 @@ protected void parseCreateField(ParseContext context) throws IOException {
         // model when ingestion starts.
         ModelMetadata modelMetadata = this.modelDao.getMetadata(modelId);
 
-        if (modelMetadata == null) {
+        if (!ModelUtil.isModelPresent(modelMetadata) || !ModelUtil.isModelCreated(modelMetadata)) {
```

Comment:
Cant this just be !ModelUtil.isModelCreated? Same as above

Reply:
We need to check if it's null first, otherwise getState will throw a null pointer exception instead of the exception we want to throw

Reply:
Unless we add the null check like your comment below says, then that would work!